cmd/root.go

```diff
@@ -274,6 +277,10 @@ func (a *AzureProvider) GetServerBank(tags map[string]string) (*ServerBank, erro
 
 		// manual filter by tag
 		for _, vm := range result.Value {
+			resourceGroupName := specifiedResourceGroupName
+			if resourceGroupName == "" {
+				resourceGroupName = extractResourceGroupName(*vm.ID) // Extract if not specified
+			}
```

Comment:
This logic will cause the ensuing `InstanceView()` call to fail if a returned instance is part of a different instance group than what is passed as a flag.